config/routes.rb

```diff
@@ -1,12 +1,16 @@
 # frozen_string_literal: true
 
 Hanami.application.routes do
-  mount :web, at: "/" do
+  # Hanami::Router#match_path? behaviour requires "/admin" to be mounted
+  # _before_ "/", otherwise requests for "/admin" still match the "/" namespace,
+  # since "/admin".start_with?("/") == true
```

Comment:
Yeah, so this is interesting. tbh I'm not sure how it was working before! But I think this change comes because of the way the router in the hanami/hanami PR enhances the `#mount` method.